client/src/components/dom/Auth/AuthModal/AuthModal.tsx

```diff
@@ -0,0 +1,246 @@
+import { useState } from "react";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import styles from "./AuthModal.module.scss";
+import { AuthAction } from "../Auth";
+
+type FormDetails = {
+  email: string;
+  password: string;
+  confirm: string;
+};
+type FormErrors = {
+  email: string;
+  password: string;
+  confirm: string;
+};
+export default function AuthModal(props: {
+  authAction: AuthAction;
+  setAuthAction: (action: AuthAction) => void;
+}) {
+  /*COMPONENT STATE*/
+  const [formDetails, setFormDetails] = useState<FormDetails>({
+    email: "",
+    password: "",
+    confirm: "",
+  });
+
+  const [formErrors, setFormErrors] = useState<FormErrors>({
+    email: "",
+    password: "",
+    confirm: "",
+  });
+  const supabase = useSupabaseClient();
+
+  if (!supabase) return;
```

Comment:
```suggestion
```

Reply:
unsure what this is intended to do, but I would not suggest having a short-circuit in the middle of a react component if avoidable.

If you want to only render under a condition, just conditionally render the JSX -- or better yet, do it in the parent component.

```
function myComponent(){
	. . .
	
	return (
		condition && { 
			<div>my component</div>
		}
	)
}
```